### PR TITLE
Initial Implementation of the Status Bar as a WASM Plugin

### LINKED ISSUE #71: Feature: Design and implement a status bar
The status bar can be something like airline, or it can be a new concept. For the time being it should display the shortcuts for the current input mode (showing the user what they can press right now in order to perform actions on the screen and updating itself accordingly).

In the future, it can also contain things like:
* Minimized panes that don't fit on screen
* List of tabs
* System information (through a plugin)
* Other ideas?

### PR DESCRIPTION
**Pull request by TheLostLambda**
Mostly just knocked out [these tasks](https://github.com/mosaic-org/mosaic/issues/71#issuecomment-756824919), which closes #71

I also patched up 23 compiler warnings and clippy lints from main, since I like clean builds!

Here is a demo video of the newly added functionality:

https://user-images.githubusercontent.com/6251883/104275412-abdcc080-549a-11eb-9c2e-60c3d5bdc43d.mp4

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 
 include!("src/cli.rs");
 
 const BIN_NAME: &str = "mosaic";
 
 fn main() {
+    // Generate Shell Completions
     let mut clap_app = CliArgs::clap();
-    println!("cargo:rerun-if-changed=src/app.rs");
+    println!("cargo:rerun-if-changed=src/cli.rs");
```
Comment (henil) [inline build.rs:12]: Oh :sweat_smile:.... Thank You!

**Comment by henil** [comment]
It is causing panic for me, whenever I try to switch mode with `Ctrl + g`

panic message: https://hatebin.com/azoyagyqdi

**Comment by TheLostLambda** [comment]
> It is causing panic for me, whenever I try to switch mode with `Ctrl + g`
> 
> panic message: https://hatebin.com/azoyagyqdi

Thanks for finding that! The bug was in the plugin code, which is now patched! The updated WASM has been shipped in this repo :)

**Comment by imsnif** [comment]
I didn't take too deep of a look at the code, but on the surface this looks great to me. Fantastic (and quick!!) work :)

**Comment by TheLostLambda** [comment]
Awesome! If it doesn't cause any issues in your testing, I'm happy merging whenever :slightly_smiling_face:

**Comment by imsnif** [comment]
Go ahead :)

**Pull request merged**